### ISSUE TITLE
todo: Complete the training and evaluation code in train.py

### ISSUE THREAD

**Issue by Shreyasvaidya** [AISocietyIITJ/NNIT#4]
Given Dataloaders and architectures, experiment with different learning rates ,architectures and also find the accuracy of prediction .(Also decoder will return vector representation of words similar to the architecture in attention is all you need paper)

Minimum Assignees - 2
Maximum Assignees- 4
